### PR TITLE
Add some internal documentation

### LINKED ISSUE #100: Document internal APIs
Currently there is no documentation for our project. The internal api should have some documentation so that it's easier to develop new features.

### PR DESCRIPTION
**Pull request by orbyfied**
- Added developer guide "How to make a feature" to `CONTRIBUTING.md`
- Added javadocs to some classes and methods

Closes #100

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -16,3 +16,144 @@ While contributing to the project, we encourage all contributors to follow good
 <br>
 
 If you have any questions or feedback, please reach out to us on Discord (link in the README).
+
+# Developer Guide: How to make a feature
```
Comment (Matthias1590) [inline CONTRIBUTING.md:20]: I kinda wanna refactor most of the code for features so I think its easiest if we leave the changes to CONTRIBUTING.md out and add them back after we're sure what the code is gonna look like. The other changes all seem fine to me

**Review by Matthias1590 — CHANGES_REQUESTED** [review]

File: src/main/java/tools/redstone/redstonetools/features/arguments/Argument.java [hunk 2]
```diff
@@ -3,12 +3,21 @@
...
+ * @param <T> The value type.
+ */
 public class Argument<T> {
+
     private String name;
     private final TypeSerializer<T, ?> type;
     private boolean optional = false;
     private T value;
     private T defaultValue;
+    // TODO: maybe add an isSet flag and unset
```
Comment (Matthias1590) [inline src/main/java/tools/redstone/redstonetools/features/arguments/Argument.java:19]: Why?

File: src/main/java/tools/redstone/redstonetools/utils/DirectionArgument.java [hunk 3]
```diff
@@ -15,10 +19,15 @@ public enum DirectionArgument {
     UP,
     DOWN,
     LEFT,
+<<<<<<< HEAD
```
Comment (Matthias1590) [inline src/main/java/tools/redstone/redstonetools/utils/DirectionArgument.java:22]: this isnt merged properly

File: CONTRIBUTING.md (hunk 1 above)
Comment (orbyfied) [reply CONTRIBUTING.md:20]: ok

File: src/main/java/tools/redstone/redstonetools/features/arguments/Argument.java (hunk 2 above)
Comment (orbyfied) [reply src/main/java/tools/redstone/redstonetools/features/arguments/Argument.java:19]: for null values

Comment (Matthias1590) [reply src/main/java/tools/redstone/redstonetools/features/arguments/Argument.java:19]: The values shouldnt be used outside of the execute method anyway

File: src/main/java/tools/redstone/redstonetools/utils/DirectionArgument.java (hunk 3 above)
Comment (orbyfied) [reply src/main/java/tools/redstone/redstonetools/utils/DirectionArgument.java:22]: oh shit

File: src/main/java/tools/redstone/redstonetools/features/arguments/Argument.java (hunk 2 above)
Comment (Matthias1590) [reply src/main/java/tools/redstone/redstonetools/features/arguments/Argument.java:19]: if the serializer accepts null or if the default is null then null is a valid value

Comment (orbyfied) [reply src/main/java/tools/redstone/redstonetools/features/arguments/Argument.java:19]: yes i know but how do we know if the value is set or not when null is a valid value

Comment (Matthias1590) [reply src/main/java/tools/redstone/redstonetools/features/arguments/Argument.java:19]: you dont need to? if the value is null its either explicitly set to null or defaulted to null, whether its set by the user or if it has that value by default shouldnt matter

**Comment by Matthias1590** [comment]
Closing this, a lot has changed since

**Pull request closed without merging**